database/engine/metadata_log/metadatalogapi.c

```diff
@@ -0,0 +1,290 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+#include "metadatalog.h"
+
+/* The buffer must not be empty */
+static void metalog_commit_record(struct metalog_instance *ctx, BUFFER *buffer, enum metalog_opcode opcode)
+{
+    struct metalog_cmd cmd;
+
+    assert(buffer_strlen(buffer));
+    assert(opcode == METALOG_COMMIT_CREATION_RECORD || opcode == METALOG_COMMIT_DELETION_RECORD);
+
+    cmd.opcode = opcode;
+    cmd.record_io_descr.buffer = buffer;
+    metalog_enq_cmd(&ctx->worker_config, &cmd);
+}
+
+static inline void metalog_commit_creation_record(struct metalog_instance *ctx, BUFFER *buffer)
+{
+    metalog_commit_record(ctx, buffer, METALOG_COMMIT_CREATION_RECORD);
+}
+
+static inline void metalog_commit_deletion_record(struct metalog_instance *ctx, BUFFER *buffer)
+{
+    metalog_commit_record(ctx, buffer, METALOG_COMMIT_DELETION_RECORD);
+}
+
+void metalog_commit_update_host(RRDHOST *host)
+{
+    struct metalog_instance *ctx;
+    BUFFER *buffer;
+
+    /* Metadata are only available with dbengine */
+    if (!host->rrdeng_ctx)
+        return;
+
+    ctx = host->rrdeng_ctx->metalog_ctx;
+    buffer = buffer_create(4096); /* This will be freed after it has been committed to the metadata log buffer */
+
+    rrdhost_rdlock(host);
+
+    buffer_sprintf(buffer,
+                   "HOST \"%s\" \"%s\" \"%s\" %d \"%s\" \"%s\" \"%s\"",
+//                 "\"%s\" \"%s\" \"%s\" \"%s\" \"%s\" \"%s\" \"%s\" \"%s\" \"%s\" \"%s\" \"%s\" \"%s\""  /* system */
+//                 "\"%s\" \"%s\" \"%s\" \"%s\" \"%s\" \"%s\" \"%s\" \"%s\" \"%s\" \"%s\" \"%s\" \"%s\"", /* info   */
+                   host->machine_guid,
+                   host->hostname,
+                   host->registry_hostname,
+                   default_rrd_update_every,
+                   host->os,
+                   host->timezone,
+                   (host->tags) ? host->tags : "");
+
+    netdata_rwlock_rdlock(&host->labels_rwlock);
+    struct label *labels = host->labels;
+    while (labels) {
+        buffer_sprintf(buffer
+            , "LABEL \"%s\" = %d %s\n"
+            , labels->key
+            , (int)labels->label_source
+            , labels->value);
+
+        labels = labels->next;
+    }
+    netdata_rwlock_unlock(&host->labels_rwlock);
+
+    buffer_strcat(buffer, "OVERWRITE labels\n");
```

Comment:
As a note - I don't think we are required to match the streaming implementation of this.

Reply:
Yes, you are right, we will fix this in https://github.com/netdata/netdata/issues/8181